### PR TITLE
[1/3 dedupe] Removes destroy modal dupe code.

### PR DESCRIPTION
**Pull request by rcorral**
This is the first of 3 PRs to consolidate the confirmation modals in the Services page.

I removed code from two different files and abstracted it into two different files, one of which (`ServiceActionModal`) will be reused for the other two modals that are remaining.

### CODE REVIEW

**Comment by rcorral** [comment]
retest this please

File: src/js/components/ServiceDetail.js [hunk 1]
```diff
@@ -11,22 +11,22 @@ import ServiceActionItem from '../constants/ServiceActionItem';
 import ServiceDetailConfigurationTab from './ServiceDetailConfigurationTab';
 import ServiceDetailDebugTab from './ServiceDetailDebugTab';
 import ServiceDetailTaskTab from './ServiceDetailTaskTab';
+import ServiceDestroyModal from './modals/ServiceDestroyModal';
```
Comment (jfurrow) [inline src/js/components/ServiceDetail.js]: :abcd:

**Comment by jfurrow** [comment]
I think this needs a rebase, I'm getting an error about the `Typeahead` not existing in `PluginModules`.

File: src/js/components/ServiceDetail.js (hunk 1 above)
Comment (rcorral) [inline src/js/components/ServiceDetail.js]: Fixed!

File: src/js/components/modals/ServiceActionModal.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  'onError'
+];
+
+/**
+ * This class is meant to be extended by Service confirmation
+ * modals, like scale and destroy.
+ */
+class ServiceActionModal extends mixin(StoreMixin) {
+  constructor() {
+    super();
```
Comment (mlunoe) [inline src/js/components/modals/ServiceActionModal.js]: `...arguments`

File: src/js/components/modals/ServiceDestroyModal.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import React from 'react';
+/* eslint-enable no-unused-vars */
+
+import MarathonStore from '../../stores/MarathonStore';
+import ServiceTree from '../../structs/ServiceTree';
+import ServiceActionModal from './ServiceActionModal';
+
+class ServiceDestroyModal extends ServiceActionModal {
+  constructor() {
+    super();
```
Comment (mlunoe) [inline src/js/components/modals/ServiceDestroyModal.js]: `...arguments`

File: src/js/components/modals/ServiceActionModal.js (hunk 2 above)
Comment (mlunoe) [inline src/js/components/modals/ServiceActionModal.js]: Fixed this for you

File: src/js/components/modals/ServiceDestroyModal.js (hunk 3 above)
Comment (mlunoe) [inline src/js/components/modals/ServiceDestroyModal.js]: Fixed this for you

**Pull request merged**